fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/logical/PushDownJoinOtherCondition.java

```diff
@@ -62,7 +64,14 @@ public Rule build() {
             if (!join.getOtherJoinCondition().isPresent()) {
                 return null;
             }
+            InferPredicate inferPredicate = new InferPredicate();
```

Comment:
Hmm. I'd suggest to adding a dedicated rule to infer predicates, rather than have side effects on predicates pushdonw rules.